### PR TITLE
[feat] Make Docker compose rely on ECR Image

### PR DESCRIPTION
**Pull request by EpicPi**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #84
* #76
* #75
* #74
* __->__ #73

## Changes made
Rewrite bin/dev
Change docker compose to use ecr image when creating containers.

## Testing
- [ ] End-to-End

## Work left to be done

### CODE REVIEW

File: bin/dev [hunk 1]
```diff
@@ -1,60 +1,71 @@
...
+        bin/util error "No container for dev found. Check that it is running with bin/dev ps, or start it with bin/dev up"
         exit 1
     fi
     eval $_resultvar="'$container'"
 }
 
-# Run docker compose commands as correct user
-runDockerCompose(){
-    user_id=$(id -u) group_id=$(id -g) docker-compose $*
+# Run commands in exiting dev container as correct user
```
Comment (SammyBaek) [inline bin/dev:17]: wdym correct user?

File: bin/dev [hunk 2]
```diff
@@ -1,60 +1,71 @@
...
+# args: commands to run in container
+runDockerExec(){
+    findBackendContainerId container_id
+    cmd="docker exec -it -u $(id -u):$(id -g) $container_id "$@
+    bin/util logAndRun $cmd
+}
+
+# Run commands with interactive turned off in exiting dev container
+# args: commands to run in container
+runDockerExecNonInteractive(){
```
Comment (SammyBaek) [inline bin/dev:27]: just curious, is there a clean way to refactor `runDockerExec` and `runDockerExecNonInteractive` into one?

File: bin/dev [hunk 3]
```diff
@@ -63,135 +74,143 @@ runWebpack(){
...
+}
 
-webpack
-    Alias for frontend.
-"
+# Hacky solution required for host machine to use python inside venv for linting python
+# Remove the symlink to py3, copy the whole py3 binary, fix the other symlinks to point to this binary
+runFixPythonLoc(){
+    runDockerExec "rm /home/worker/app/server/.venv/bin/python3.8"
+    runDockerExec "cp /usr/local/bin/python3.8 /home/worker/app/server/.venv/bin/python3.8"
```
Comment (SammyBaek) [inline bin/dev:127]: lol how long does this take to cp py3 binary?

Comment (EpicPi) [reply bin/dev:127]: basically zero time

File: bin/dev (hunk 2 above)
Comment (EpicPi) [reply bin/dev:27]: There is, but the shell syntax makes it really verbose, and then calling it also becomes a pain.

File: bin/dev (hunk 1 above)
Comment (EpicPi) [reply bin/dev:17]: Unreleated to ecr login. 
This is the user inside of the docker container. Docker falls back to using root, but that means if docker writes files to your disk, those files are still owned by root.

**Review by SammyBaek — APPROVED** [review]

**Pull request merged**